### PR TITLE
Support AWS

### PR DESCRIPTION
**Pull request by milliams**

### CODE REVIEW

**Comment by milliams** [comment]
I've run the tests on AWS and Oracle and they pass without issue. I'd like to merge this in so that we can continue development. AWS support is still not supported fully and there's no documentation. We also have no compatibility promise for AWS clusters.

See also ACRC/slurm-ansible-playbook#70 for the ansible changes.

**Comment by christopheredsall** [comment]
Works well for my manual test cluster.

LGTM +1

File: aws/networking.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  enable_dns_hostnames = "true"
+  tags = {
+    Name = "citc-net-${local.cluster_id}"
+    cluster = local.cluster_id
+  }
+}
+
+resource "aws_subnet" "vpc_subnetwork" {
+  vpc_id     = aws_vpc.vpc_network.id
+  cidr_block = "10.0.1.0/24"
```
Comment (christopheredsall) [inline aws/networking.tf]: Is this a limit to the size of the cluster? a /24 is only 256 addresses, 254 usable.

File: aws/provider.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Pin the terraform and provider versions to avoid issues if the
```
Comment (christopheredsall) [inline aws/provider.tf]: Comment is missing the end of the sentence.

File: aws/networking.tf (hunk 1 above)
Comment (milliams) [reply aws/networking.tf]: I think it was arbitrarily chosen. I'll increase it to a `/16` or a `/8`. I assume that `10.0.0.0/8` is possible?

Comment (christopheredsall) [reply aws/networking.tf]: I think the subnet would need to be smaller than the containing network (currently /16)

Comment (milliams) [reply aws/networking.tf]: I've made the VPC a `10.0.0.0/8` and the subnet a `10.1.0.0/16` for now. This matches Google and Oracle.

Comment (milliams) [reply aws/networking.tf]: It seems that AWS has [a limit](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#VPC_Sizing) of `/16` for a VPC. I'll make the VPC `10.0.0.0/16` and the subnet `10.0.0.0/17` for now.

**Pull request merged**